### DISCUSSION TITLE
What's the equivalent of setfilter in V8? (Table API)

### DISCUSSION THREAD

**Discussion by decayedCell** [TanStack/table#4316] [Q&A] [answered]
So I'm trying to filter my table by at least 2 columns for example:
in v7:
```
setFilter('country', 'Italy');
setFilter('areaType', 'rural');
```
In v8 I tried to do 
```
table.setColumnFilters([{ 'country', 'Italy' }]);
table.setColumnFilters([{ 'areaType', 'rural' }]);
```
But this overwrites the whole array instead of adding a 2nd way to filter.
I know in the example ```column.setFilterValue``` is being used but I was wondering if there's a more table-ish way to do this similar to v7.

**Comment by decayedCell** [comment, marked as answer]
Just after asking this I found a nice solution for what I was trying to do
```
...
table.getAllColumns().forEach((column) => {
    if (column.id === 'country') {
      column.setFilterValue('Italy');
    }
});
...
````

**Reply by alexandre-embie** [reply]
Is there anything else to do to make this work ? I'm trying to do the same and the second column id filter does not get triggered, I think because the first one is not returning true, so the second one does not get applied
